src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -291,6 +291,36 @@ public void testGVCFModeIsConcordantWithGATK3_8AlelleSpecificResults(final Strin
         Assert.assertTrue(concordance >= 0.99, "Concordance with GATK 3.8 in AS GVCF mode is < 99% (" +  concordance + ")");
     }
 
+    @Test
+    public void testGVCFModeGenotypePosteriors() throws Exception {
+        Utils.resetRandomGenerator();
+
+        final String inputFileName = NA12878_20_21_WGS_bam;
+        final String referenceFileName =b37_reference_20_21;
+
+        final File output = createTempFile("testGVCFModeIsConsistentWithPastResults", ".g.vcf");
+        final File expected = new File(TEST_FILES_DIR, "expected.testGVCFMode.gatk4.posteriors.g.vcf");
+
+        final String[] args = {
+                "-I", inputFileName,
+                "-R", referenceFileName,
+                "-L", "20:10000000-10100000",
+                "-O", output.getAbsolutePath(),
+                "-ERC", "GVCF",
+                "--apply-priors",
+                "-supporting", largeFileTestDir + "1000G.phase3.broad.withGenotypes.chr20.10100000.vcf",
+                "--num-reference-samples-if-no-call", "2500",
+                "-pairHMM", "AVX_LOGLESS_CACHING",
+                "-lenient",
+                "--" + StandardArgumentDefinitions.ADD_OUTPUT_VCF_COMMANDLINE, "false"
+        };
+
+        runCommandLine(args);
+
+        // Test for an exact match against past results
+        IntegrationTestSpec.assertEqualTextFiles(output, expected);
```

Comment:
Can you think of a way to construct a good test that doesn't require an exact match on the output file? I've been trying to move `HaplotypeCallerIntegrationTest` away from that style of test. Eg., could you call `VariantContextTestUtils.readEntireVCFIntoMemory()` and inspect the VCs themselves?

Reply:
Sure.  I can refactor the integration test in the same way as in PosteriorProbabilitiesUtilsUnitTest.